### PR TITLE
KIALI-3139 Use internal url to determine if kiali user has access to jaeger

### PR DESCRIPTION
**Pull request by xeviknal**
** Describe the change **

In order to determine if kiali user has access to jaeger, we need to use internal urls

** Issue reference **
https://issues.jboss.org/browse/KIALI-3139

** Backwards incompatible? **
yes

### CODE REVIEW

File: business/jaeger_helper.go [hunk 1]
```diff
@@ -77,6 +77,10 @@ func getErrorTracesFromJaeger(namespace string, service string, requestToken str
 	return errorTraces, err
 }
 
+func GetJaegerInternalURL(path string) (*url.URL, error) {
+	return url.Parse(fmt.Sprintf("http://%s%s%s", appstate.JaegerConfig.Service, appstate.JaegerConfig.Path, path))
```
Comment (jotak) [inline business/jaeger_helper.go]: The URL is built from service without port ... is that supposed to work ?
For me it seems it doesn't. E.g. from kiali pod I have to curl on `http://jaeger-query:16686/...` and not just `http://jaeger-query/...`

**Comment by aljesusg** [comment]
Screenshot from 2019-07-18 15-20-52

**Comment by xeviknal** [comment]
@aljesusg does your jaeger-query service runs in port 80? I think maistra 0.11 changes the jaeger tracing to use port 80. That is why you could fetch jaeger without the port.
In maistra 12, jaeger is under 443 (https). And istio mainstream uses 16646 port.
So I understand that the traces count in the traces tab is only working with jaeger under port 80.
https://github.com/kiali/kiali/blob/v1.1/business/jaeger_helper.go#L43

I propose to autodiscover the port in the autodiscovery process.

I did fill in the JIRA: https://issues.jboss.org/browse/KIALI-3151

**Comment by aljesusg** [comment]
@xeviknal tests that we did in the bluejeans were in  GKE with Istio 1.1.9 , it's not related with Maistra that's the point ^^

**Comment by hhovsepy** [comment]
Verified, when Jeager config refers to a namespace Kiali does not have access.
Screenshot from 2019-07-26 13-37-23

**Pull request merged**